browser/ui/views/brave_shields/cookie_consent_browser_helper.cc

```diff
@@ -0,0 +1,218 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/brave_shields/cookie_consent_browser_helper.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/containers/flat_set.h"
+#include "base/feature_list.h"
+#include "base/numerics/clamped_math.h"
+#include "brave/browser/brave_browser_process.h"
+#include "brave/browser/ui/webui/brave_shields/cookie_consent_blocker_ui.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/common/brave_shield_constants.h"
+#include "brave/components/brave_shields/common/features.h"
+#include "brave/components/brave_shields/common/pref_names.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/views/bubble/webui_bubble_manager.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "chrome/browser/ui/views/location_bar/location_bar_view.h"
+#include "components/grit/brave_components_strings.h"
+#include "content/public/browser/web_contents.h"
+
+namespace brave_shields {
+
+namespace {
+
+// The number of unique host names loaded as the primary page in the active tab
+// before we show the cookie consent blocker bubble.
+constexpr int kTabLoadThreshold = 10;
+
+views::View* GetAnchorView(Browser* browser) {
+  DCHECK(browser);
+  auto* browser_view = BrowserView::GetBrowserViewForBrowser(browser);
+  DCHECK(browser_view);
+  auto* location_bar_view = browser_view->GetLocationBarView();
+  if (location_bar_view) {
+    return location_bar_view;
+  }
+  return browser_view;
+}
+
+class CookieConsentProfileHelper : public base::SupportsUserData::Data {
```

Comment:
this seems a bit problematic because unfortunately the list selections are global. If the tab count resets on browser restart then use a transient static counter, otherwise I would store the count in local state as well

Reply:
The "host set" is now a static var.